agent/nodeattestor_test.go

```diff
@@ -1,18 +1,301 @@
-package main_test
+package main
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/spiffe/spire-plugin-sdk/pluginsdk"
 	"github.com/spiffe/spire-plugin-sdk/plugintest"
+	agentnodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
 	nodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
 	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
 	"github.com/spiffe/spire-plugin-sdk/templates/agent/nodeattestor"
+	"github.com/spiffe/spire/pkg/common/pemutil"
+	"github.com/spiffe/spire/pkg/common/plugin/k8s"
+	sat_common "github.com/spiffe/spire/pkg/common/plugin/k8s"
+	"github.com/spiffe/spire/test/spiretest"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+var sampleKeyPEM = []byte(`-----BEGIN RSA PRIVATE KEY-----
+MIIBywIBAAJhAMB4gbT09H2RKXaxbu6IV9C3WY+pvkGAbrlQRIHLHwV3Xt1HchjX
+c08v1VEoTBN2YTjhZJlDb/VUsNMJsmBFBBted5geRcbrDtXFlUJ8tQoQx1dWM4Aa
+xcdULJ83A9ICKwIDAQABAmBR1asInrIphYQEtHJ/NzdnRd3tqHV9cjch0dAfA5dA
+Ar4yBYOsrkaX37WqWSDnkYgN4FWYBWn7WxeotCtA5UQ3SM5hLld67rUqAm2dLrs1
+z8va6SwLzrPTu2+rmRgovFECMQDpbfPBRex7FY/xWu1pYv6X9XZ26SrC2Wc6RIpO
+38AhKGjTFEMAPJQlud4e2+4I3KkCMQDTFLUvBSXokw2NvcNiM9Kqo5zCnCIkgc+C
+hM3EzSh2jh4gZvRzPOhXYvNKgLx8+LMCMQDL4meXlpV45Fp3eu4GsJqi65jvP7VD
+v1P0hs0vGyvbSkpUo0vqNv9G/FNQLNR6FRECMFXEMz5wxA91OOuf8HTFg9Lr+fUl
+RcY5rJxm48kUZ12Mr3cQ/kCYvftL7HkYR/4rewIxANdritlIPu4VziaEhYZg7dvz
+pG3eEhiqPxE++QHpwU78O+F1GznOPBvpZOB3GfyjNQ==
+-----END RSA PRIVATE KEY-----`)
+
+const (
+		podName = "MYPOD"
+		clusterName = "MYCLUSTER"
+)
+
+type attestorSuite struct {
+	agentPlugin         *Plugin
+	agentAttestorClient *agentnodeattestorv1.NodeAttestorPluginClient
+	agentHCL            string
+
+	psatData  *k8s.PSATAttestationData
+	token     string
+	tokenPath string
+
+	t       *testing.T
+	require *require.Assertions
+}
+
+func (a *attestorSuite) loadAgentPlugin(agentHLC string) error {
+	a.agentPlugin = new(Plugin)
+
+	a.agentAttestorClient = new(agentnodeattestorv1.NodeAttestorPluginClient)
+	configClient := new(configv1.ConfigServiceClient)
+
+	plugintest.ServeInBackground(a.t, plugintest.Config{
+		PluginServer:   agentnodeattestorv1.NodeAttestorPluginServer(a.agentPlugin),
+		PluginClient:   a.agentAttestorClient,
+		ServiceServers: []pluginsdk.ServiceServer{configv1.ConfigServiceServer(a.agentPlugin)},
+		ServiceClients: []pluginsdk.ServiceClient{configClient},
+	})
+
+	hcl := fmt.Sprintf(`
+		cluster = "FOO"
+		token_path = %q
+	`, a.tokenPath)
+
+	if agentHLC != "" {
+		hcl = agentHLC
+	}
+
+	_, err := configClient.Configure(context.Background(), &configv1.ConfigureRequest{
+		HclConfiguration: hcl,
+		CoreConfiguration: &configv1.CoreConfiguration{
+			TrustDomain: "TrustDomain",
+		},
+	})
+
+	return err
+}
+
+func (a *attestorSuite) createAndWriteToken() {
+	var err error
+	dir := "/temp/token"
```

Comment:
```suggestion
	dir := "/tmp/token"
```
The default temp dir for linux is /tmp/